provider/hcloud/hcloud_discover.go

```diff
@@ -0,0 +1,130 @@
+// Package hcloud provides node discovery for Hetzner Cloud.
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+type Provider struct{}
+
+func (p *Provider) Help() string {
+	return `Hetzner Cloud:
+
+	provider:       "hcloud"
+	location:       The Hetzner Cloud datacenter location to filter by (eg. "fsn1")
+	label_selector: The label selector to filter by
+	address_type:   "private_v4", "public_v4" or "public_v6", defaults to "private_v4". In the case of private networks, the first one will be used
+	api_token:      The Hetzner Cloud API token to use, can also be provided by environment variable: HCLOUD_TOKEN
+`
+}
+
+// getServerIP returns the IP address of the specified type for the hcloud server.
+func getServerIP(s *hcloud.Server, addrType string, l *log.Logger) string {
```

Comment:
```suggestion
// serverIP returns the IP address of the specified type for the hcloud server.
func serverIP(s *hcloud.Server, addrType string, l *log.Logger) string {
```

`get` prefix is not idiomatic: https://golang.org/doc/effective_go#Getters

Reply:
thanks! this is fixed.